datafusion/expr-common/src/interval_arithmetic.rs

```diff
@@ -1119,11 +1180,11 @@ fn next_value_helper<const INC: bool>(value: ScalarValue) -> ScalarValue {
     match value {
         // f32/f64::NEG_INF/INF and f32/f64::NaN values should not emerge at this point.
         Float32(Some(val)) => {
-            assert!(val.is_finite(), "Non-standardized floating point usage");
+            debug_assert!(val.is_finite(), "Non-standardized floating point usage");
```

Comment:
This will strip the assert away from production build, is this intended?

Reply:
In general, I don't like to waste compute cycles on such impossibility assertions in production builds when writing low-level code (like interval arithmetic), which can run very frequently. On the other hand, in high-level code that doesn't execute very often, using ordinary assertions is perfectly fine.